### PR TITLE
tox: add generic xdist factor

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Motivated by using `-xdist` for more jobs (for faster builds).
But then I've noticed that we run both `py37` and `py37-xdist` also..

Is it sensible to use `-xdist` with py34, py35, and py36?

File: tox.ini [hunk 1]
```diff
@@ -20,18 +20,21 @@ envlist =
 
 [testenv]
 commands =
-    {env:_PYTEST_TOX_COVERAGE_RUN:} pytest --lsof {posargs}
+    {env:_PYTEST_TOX_COVERAGE_RUN:} pytest {env:_PYTEST_TOX_ARGS} {posargs}
     coverage: coverage combine
     coverage: coverage report
 passenv = USER USERNAME COVERAGE_* TRAVIS
 setenv =
+    _PYTEST_TOX_ARGS=--lsof
```
Comment (RonnyPfannschmidt) [inline tox.ini]: i suspect this needs a comment that taking off lsoff for xdist is intended

we still need some thing to run lsoff against tho

**Comment by nicoddemus** [comment]
> Is it sensible to use `-xdist` with py34, py35, and py36?

I don't think it is necessary... we run xdist to ensure we are not breaking things with it, and we run both `py27-xdist` and `py37-xdist` to catch py2 vs py3 discrepancies.

**Comment by blueyed** [comment]
> > Is it sensible to use `-xdist` with py34, py35, and py36?
> 
> I don't think it is necessary...

My intention is to make the builds faster.

I suggest to use `-xdist` for all normal (`py*`) builds, but then maybe py27 and py37 additionally to test them without xdist.

**Comment by nicoddemus** [comment]
> I suggest to use `-xdist` for all normal (`py*`) builds, but then maybe py27 and py37 additionally to test them without xdist.

Oh sorry, I didn't catch that meaning at first. I think that's fair. 👍

File: .travis.yml [hunk 2]
```diff
@@ -32,8 +32,10 @@ jobs:
   include:
     # Coverage tracking is slow with pypy, skip it.
     - env: TOXENV=pypy-xdist PYTEST_NO_COVERAGE=1
-      python: 'pypy-5.4'
-      dist: trusty
+      python: 'pypy2.7-6.0'
+    - env: TOXENV=pypy3-xdist PYTEST_NO_COVERAGE=1
+      python: 'pypy3.5-6.0'
+
```
Comment (blueyed) [inline .travis.yml]: Relevant: https://github.com/pytest-dev/pytest/issues/3460.

File: .travis.yml [hunk 3]
```diff
@@ -31,16 +31,16 @@ matrix:
 jobs:
   include:
     # Coverage tracking is slow with pypy, skip it.
-    - env: TOXENV=pypy PYTEST_NO_COVERAGE=1
+    - env: TOXENV=pypy-xdist PYTEST_NO_COVERAGE=1
```
Comment (RonnyPfannschmidt) [inline .travis.yml]: i believe we need to ensure lsof on pypy due to gc differences in order to catch unwanted issues

Comment (blueyed) [reply .travis.yml]: Ok.

This is not related to pypy-xdist failing on Windows though, is it?
https://ci.appveyor.com/project/pytestbot/pytest/builds/22239624/job/2hns6pgk71weay26

File: tox.ini (hunk 1 above)
Comment (blueyed) [reply tox.ini]: `--lsof` was not used with the previous `py27-xdist` / `py37-xdist` jobs, that's why I've removed it here.  Will try to keep it in general.

File: .travis.yml (hunk 3 above)
Comment (RonnyPfannschmidt) [reply .travis.yml]: no, thats probably an issue with execnet

**Comment by blueyed** [comment]
Reverted this back to only change tox.ini for now.

**Pull request merged**